### ISSUE TITLE
Review repo PR checks

### ISSUE THREAD

**Issue by devlace** [Azure-Samples/modern-data-warehouse-dataops#771]
DoD
- Revisit automatic github PR checks (ie. Linting). Make a decision if need to add additional checks.

**Comment by naga-nandyala** [comment]
I have started working on this.
Few findings:
----------------------

1. I have seen that Formatting, Linting, checks are not happening in our repo in a conventional way. 
2. They are happening in the code during the generation of build process.
3. Though few checks are configured at the repo level (mde repo), they are not getting triggered (e.g. Terraform related checks.)

Proposal/Way forward 
------------
1. Look holistically how we want our repo to have these checks (formatting, linting, other pre-commit checks).
2. Try these within Fabric E2E sample.
3. Rest of the code (e.g. databricks, single tech samples) 

Current Progress:
-------------
1. I have created a personal repo to demonstrate how this can be done.  (this was demonstrated to Anuj and Lace)
2. I am currently in process of adding these to MDW Fabric E2E samples.
